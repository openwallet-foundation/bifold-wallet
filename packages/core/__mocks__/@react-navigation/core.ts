const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

export { useNavigation, useIsFocused, CommonActions }
