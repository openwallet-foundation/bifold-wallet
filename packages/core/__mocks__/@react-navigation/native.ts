const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const useRoute = jest.fn().mockReturnValue({ params: {} })

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useNavigation, useIsFocused, useRoute, useFocusEffect, createNavigatorFactory, CommonActions }
