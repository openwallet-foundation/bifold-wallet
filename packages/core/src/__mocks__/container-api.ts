import { BrandingOverlayType, DefaultOCABundleResolver } from '@hyperledger/aries-oca/build/legacy'

const { TOKENS: T } = jest.requireActual('../../src/container-api')

export { T as TOKENS }

// eslint-disable-next-line @typescript-eslint/no-var-requires
const resolver = new DefaultOCABundleResolver(require('../../src/assets/oca-bundles.json'), {
  brandingOverlayType: BrandingOverlayType.Branding10,
})

export const useContainer = jest.fn().mockReturnValue({
  resolve: (token: typeof T) => {
    switch (token) {
      case T.UTIL_OCA_RESOLVER:
        return resolver
      case T.CRED_HELP_ACTION_OVERRIDES:
        return []
      case T.NOTIFICATIONS:
        return { useNotifications: jest.fn(), customNotificationConfig: undefined }
      default:
        return undefined
    }
  },
})

// export const useContainer = jest.fn().mockReturnValue({
//   resolve: jest.fn().mockReturnValue({
//     resolve: jest.fn().mockImplementation(() => Promise.resolve({})),
//     resolveAllBundles: jest.fn().mockImplementation(() => Promise.resolve({})),
//   }),
// })
