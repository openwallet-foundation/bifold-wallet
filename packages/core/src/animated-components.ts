import ButtonLoading from './components/animated/ButtonLoading'
import ConnectionLoading from './components/animated/ConnectionLoading'
import CredentialAdded from './components/animated/CredentialAdded'
import CredentialPending from './components/animated/CredentialPending'
import LoadingIndicator from './components/animated/LoadingIndicator'
import RecordLoading from './components/animated/RecordLoading'
import SendingProof from './components/animated/SendingProof'
import SentProof from './components/animated/SentProof'

export interface AnimatedComponents {
  ButtonLoading: React.FC
  ConnectionLoading: React.FC
  CredentialAdded: React.FC
  CredentialPending: React.FC
  LoadingIndicator: React.FC
  RecordLoading: React.FC
  SendingProof: React.FC
  SentProof: React.FC
}

export const animatedComponents: AnimatedComponents = {
  ButtonLoading: ButtonLoading,
  ConnectionLoading: ConnectionLoading,
  CredentialAdded: CredentialAdded,
  CredentialPending: CredentialPending,
  LoadingIndicator: LoadingIndicator,
  RecordLoading: RecordLoading,
  SendingProof: SendingProof,
  SentProof: SentProof,
}
