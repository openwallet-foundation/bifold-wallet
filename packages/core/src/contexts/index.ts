import * as reducers from './reducers'
import * as store from './store'
export { reducers, store }
