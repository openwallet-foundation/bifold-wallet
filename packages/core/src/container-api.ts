import { Agent } from '@credo-ts/core'
import { IndyVdrPoolConfig } from '@credo-ts/indy-vdr'
import { ProofRequestTemplate } from '@hyperledger/aries-bifold-verifier'
import { OCABundleResolverType } from '@hyperledger/aries-oca/build/legacy'
import { StackNavigationProp } from '@react-navigation/stack'
import React, { createContext, useContext } from 'react'
import { DependencyContainer } from 'tsyringe'

import { Button } from './components/buttons/Button-api'
import { ReducerAction } from './contexts/reducers/store'
import { IHistoryManager } from './modules/history'
import Onboarding from './screens/Onboarding'
import UpdateAvailable from './screens/UpdateAvailable'
import { AttestationMonitor } from './types/attestation'
import { IVersionCheckService } from './types/version-check'
import { GenericFn } from './types/fn'
import { AuthenticateStackParams, ScreenLayoutConfig, ScreenOptionsType, OnboardingTask } from './types/navigators'
import { CustomNotification } from './types/notification'
import { Config, HistoryEventsLoggerConfig } from './types/config'
import { State } from './types/state'
import { NotificationReturnType, NotificationsInputProps } from './hooks/notifications'
import { NotificationListItemProps } from './components/listItems/NotificationListItem'
import { PINCreateHeaderProps } from './components/misc/PINCreateHeader'
import { PINExplainerProps } from './screens/PINExplainer'
import { CredentialListFooterProps } from './types/credential-list-footer'
import { ContactListItemProps } from './components/listItems/ContactListItem'
import { ContactCredentialListItemProps } from './components/listItems/ContactCredentialListItem'
import { InlineErrorConfig } from './types/error'
import { BifoldLogger } from './services/logger'

export type FN_ONBOARDING_DONE = (
  dispatch: React.Dispatch<ReducerAction<unknown>>,
  navigation: StackNavigationProp<AuthenticateStackParams>
) => GenericFn

type LoadStateFn = (dispatch: React.Dispatch<ReducerAction<unknown>>) => Promise<void>

type GenerateOnboardingWorkflowStepsFn = (state: State, config: Config, termsVersion: number) => Array<OnboardingTask>

type ProofRequestTemplateFn = (useDevTemplates: boolean) => Array<ProofRequestTemplate>

export const PROOF_TOKENS = {
  GROUP_BY_REFERENT: 'proof.groupByReferant',
  CRED_HELP_ACTION_OVERRIDES: 'proof.credHelpActionOverride',
} as const

export const SCREEN_TOKENS = {
  SCREEN_PREFACE: 'screen.preface',
  SCREEN_UPDATE_AVAILABLE: 'screen.update-available',
  SCREEN_TERMS: 'screen.terms',
  SCREEN_ONBOARDING: 'screen.onboarding',
  SCREEN_DEVELOPER: 'screen.developer',
  SCREEN_ONBOARDING_ITEM: 'screen.onboarding.item',
  SCREEN_ONBOARDING_PAGES: 'screen.onboarding.pages',
  SCREEN_SPLASH: 'screen.splash',
  SCREEN_SCAN: 'screen.scan',
  SCREEN_USE_BIOMETRY: 'screen.use-biometry',
  SCREEN_PIN_EXPLAINER: 'screen.pin-explainer',
} as const

export const NAV_TOKENS = {
  CUSTOM_NAV_STACK_1: 'nav.slot1',
} as const

export const COMPONENT_TOKENS = {
  COMPONENT_HOME_HEADER: 'component.home.header',
  COMPONENT_HOME_NOTIFICATIONS_EMPTY_LIST: 'component.home.notifications-empty-list',
  COMPONENT_HOME_FOOTER: 'component.home.footer',
  COMPONENT_CRED_EMPTY_LIST: 'component.cred.empty-list',
  COMPONENT_RECORD: 'component.record',
  COMPONENT_PIN_CREATE_HEADER: 'component.pin-create-header',
  COMPONENT_CONTACT_LIST_ITEM: 'component.contact-list-item',
  COMPONENT_CONTACT_DETAILS_CRED_LIST_ITEM: 'component.contact-details-cred-list-item',
  COMPONENT_CONNECTION_ALERT: 'component.connection-alert',
} as const

export const NOTIFICATION_TOKENS = {
  NOTIFICATIONS: 'notification.list',
  NOTIFICATIONS_LIST_ITEM: 'notification.list-item',
} as const

export const STACK_TOKENS = {
  STACK_ONBOARDING: 'stack.onboarding',
} as const

export const FN_TOKENS = {
  FN_ONBOARDING_DONE: 'fn.onboardingDone',
  COMPONENT_CRED_LIST_HEADER_RIGHT: 'fn.credListHeaderRight',
  COMPONENT_CRED_LIST_OPTIONS: 'fn.credListOptions',
  COMPONENT_CRED_LIST_FOOTER: 'fn.credListFooter',
} as const

export const HISTORY_TOKENS = {
  FN_LOAD_HISTORY: 'fn.loadHistory',
  HISTORY_ENABLED: 'history.enabled',
  HISTORY_EVENTS_LOGGER: 'history.eventsLogger',
} as const

export const COMP_TOKENS = {
  COMP_BUTTON: 'comp.button',
} as const

export const SERVICE_TOKENS = {
  SERVICE_TERMS: 'screen.terms',
} as const

export const LOAD_STATE_TOKENS = {
  LOAD_STATE: 'state.load',
} as const

export const OBJECT_TOKENS = {
  OBJECT_SCREEN_CONFIG: 'object.screen-config',
  OBJECT_LAYOUT_CONFIG: 'object.screenlayout-config',
} as const

export const CACHE_TOKENS = {
  CACHE_CRED_DEFS: 'cache.cred-defs',
  CACHE_SCHEMAS: 'cache.schemas',
} as const

export const UTILITY_TOKENS = {
  UTIL_LOGGER: 'utility.logger',
  UTIL_OCA_RESOLVER: 'utility.oca-resolver',
  UTIL_LEDGERS: 'utility.ledgers',
  UTIL_PROOF_TEMPLATE: 'utility.proof-template',
  UTIL_ATTESTATION_MONITOR: 'utility.attestation-monitor',
  UTIL_APP_VERSION_MONITOR: 'utility.app-version-monitor',
} as const

export const CONFIG_TOKENS = {
  CONFIG: 'config',
  INLINE_ERRORS: 'errors.inline',
  ONBOARDING: 'utility.onboarding',
} as const

export const TOKENS = {
  ...PROOF_TOKENS,
  ...COMPONENT_TOKENS,
  ...SCREEN_TOKENS,
  ...NAV_TOKENS,
  ...SERVICE_TOKENS,
  ...STACK_TOKENS,
  ...NOTIFICATION_TOKENS,
  ...FN_TOKENS,
  ...COMP_TOKENS,
  ...LOAD_STATE_TOKENS,
  ...OBJECT_TOKENS,
  ...CACHE_TOKENS,
  ...UTILITY_TOKENS,
  ...CONFIG_TOKENS,
  ...HISTORY_TOKENS,
} as const

export type FN_HISTORY_MANAGER = (agent: Agent<any>) => IHistoryManager

export type TokenMapping = {
  [TOKENS.CRED_HELP_ACTION_OVERRIDES]: {
    credDefIds: string[]
    schemaIds: string[]
    action: (navigation: any) => void
  }[]
  [TOKENS.GROUP_BY_REFERENT]: boolean
  [TOKENS.SCREEN_PREFACE]: React.FC
  [TOKENS.SCREEN_UPDATE_AVAILABLE]: typeof UpdateAvailable
  [TOKENS.STACK_ONBOARDING]: React.FC
  [TOKENS.SCREEN_TERMS]: { screen: React.FC; version: boolean | string }
  [TOKENS.SCREEN_DEVELOPER]: React.FC
  [TOKENS.SCREEN_ONBOARDING_PAGES]: (onTutorialCompleted: GenericFn, OnboardingTheme: any) => Array<Element>
  [TOKENS.SCREEN_SPLASH]: React.FC
  [TOKENS.SCREEN_SCAN]: React.FC
  [TOKENS.SCREEN_USE_BIOMETRY]: React.FC
  [TOKENS.SCREEN_ONBOARDING]: typeof Onboarding
  [TOKENS.SCREEN_PIN_EXPLAINER]: React.FC<PINExplainerProps>
  [TOKENS.FN_ONBOARDING_DONE]: FN_ONBOARDING_DONE
  [TOKENS.LOAD_STATE]: LoadStateFn
  [TOKENS.COMP_BUTTON]: Button
  [TOKENS.NOTIFICATIONS]: {
    useNotifications: ({ openIDUri }: NotificationsInputProps) => NotificationReturnType
    customNotificationConfig?: CustomNotification
  }
  [TOKENS.NOTIFICATIONS_LIST_ITEM]: React.FC<NotificationListItemProps>
  [TOKENS.OBJECT_SCREEN_CONFIG]: ScreenOptionsType
  [TOKENS.OBJECT_LAYOUT_CONFIG]: ScreenLayoutConfig
  [TOKENS.COMPONENT_PIN_CREATE_HEADER]: React.FC<PINCreateHeaderProps>
  [TOKENS.CACHE_CRED_DEFS]: { did: string; id: string }[]
  [TOKENS.CACHE_SCHEMAS]: { did: string; id: string }[]
  [TOKENS.UTIL_LOGGER]: BifoldLogger
  [TOKENS.UTIL_OCA_RESOLVER]: OCABundleResolverType
  [TOKENS.UTIL_LEDGERS]: IndyVdrPoolConfig[]
  [TOKENS.UTIL_PROOF_TEMPLATE]: ProofRequestTemplateFn | undefined
  [TOKENS.UTIL_ATTESTATION_MONITOR]: AttestationMonitor
  [TOKENS.UTIL_APP_VERSION_MONITOR]: IVersionCheckService
  [TOKENS.FN_LOAD_HISTORY]: FN_HISTORY_MANAGER
  [TOKENS.HISTORY_ENABLED]: boolean
  [TOKENS.HISTORY_EVENTS_LOGGER]: HistoryEventsLoggerConfig
  [TOKENS.CONFIG]: Config
  [TOKENS.ONBOARDING]: GenerateOnboardingWorkflowStepsFn
  [TOKENS.COMPONENT_CRED_LIST_HEADER_RIGHT]: React.FC
  [TOKENS.COMPONENT_CRED_LIST_OPTIONS]: React.FC
  [TOKENS.COMPONENT_CRED_LIST_FOOTER]: React.FC<CredentialListFooterProps>
  [TOKENS.COMPONENT_HOME_HEADER]: React.FC
  [TOKENS.COMPONENT_HOME_NOTIFICATIONS_EMPTY_LIST]: React.FC
  [TOKENS.COMPONENT_HOME_FOOTER]: React.FC
  [TOKENS.COMPONENT_CRED_EMPTY_LIST]: React.FC
  [TOKENS.COMPONENT_RECORD]: React.FC
  [TOKENS.COMPONENT_CONTACT_LIST_ITEM]: React.FC<ContactListItemProps>
  [TOKENS.COMPONENT_CONTACT_DETAILS_CRED_LIST_ITEM]: React.FC<ContactCredentialListItemProps>
  [TOKENS.INLINE_ERRORS]: InlineErrorConfig
  [TOKENS.CUSTOM_NAV_STACK_1]: React.FC
  [TOKENS.COMPONENT_CONNECTION_ALERT]: React.FC<{ connectionID?: string }>
}

export interface Container {
  init(): Container
  resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K]
  resolveAll<K extends keyof TokenMapping, T extends K[]>(tokens: [...T]): { [I in keyof T]: TokenMapping[T[I]] }
  get container(): DependencyContainer
}

export const ContainerContext = createContext<Container | undefined>(undefined)

export const ContainerProvider = ContainerContext.Provider

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
export const useContainer = () => useContext(ContainerContext)!

export const useServices = <K extends keyof TokenMapping, T extends K[]>(tokens: [...T]) => {
  return useContainer().resolveAll(tokens)
}
