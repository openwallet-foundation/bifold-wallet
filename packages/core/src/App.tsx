import React, { useEffect } from 'react'
import { StatusBar } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import { useNavigationContainerRef } from '@react-navigation/native'
import { isTablet } from 'react-native-device-info'
import Orientation from 'react-native-orientation-locker'
import { animatedComponents } from './animated-components'
import ErrorModal from './components/modals/ErrorModal'
import toastConfig from './components/toast/ToastConfig'
import { tours } from './constants'
import { Container, ContainerProvider } from './container-api'
import { AnimatedComponentsProvider } from './contexts/animated-components'
import { AuthProvider } from './contexts/auth'
import NavContainer from './contexts/navigation'
import { NetworkProvider } from './contexts/network'
import { StoreProvider } from './contexts/store'
import { ThemeProvider } from './contexts/theme'
import { TourProvider } from './contexts/tour/tour-provider'
import { initStoredLanguage } from './localization'
import RootStack from './navigators/RootStack'
import { bifoldTheme, themes } from './theme'

const createApp = (container: Container): React.FC => {
  const AppComponent: React.FC = () => {
    const navigationRef = useNavigationContainerRef()

    useEffect(() => {
      initStoredLanguage().then()
    }, [])

    useEffect(() => {
      // Hide the native splash / loading screen so that our
      // RN version can be displayed.
      SplashScreen.hide()
    }, [])

    if (!isTablet()) {
      Orientation.lockToPortrait()
    }

    return (
      <ContainerProvider value={container}>
        <StoreProvider>
          <ThemeProvider themes={themes} defaultThemeName={bifoldTheme.themeName}>
            <NavContainer navigationRef={navigationRef}>
              <AnimatedComponentsProvider value={animatedComponents}>
                <AuthProvider>
                  <NetworkProvider>
                    <StatusBar
                      hidden={false}
                      barStyle="light-content"
                      backgroundColor={bifoldTheme.ColorPalette.brand.primary}
                      translucent={false}
                    />
                    <ErrorModal />
                    <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
                      <RootStack />
                    </TourProvider>
                    <Toast topOffset={15} config={toastConfig} />
                  </NetworkProvider>
                </AuthProvider>
              </AnimatedComponentsProvider>
            </NavContainer>
          </ThemeProvider>
        </StoreProvider>
      </ContainerProvider>
    )
  }

  return AppComponent
}

export default createApp
