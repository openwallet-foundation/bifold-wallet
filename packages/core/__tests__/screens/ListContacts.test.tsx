import { fireEvent, render, waitFor } from '@testing-library/react-native'
import React from 'react'
import { container } from 'tsyringe'

import { useNavigation as testUseNavigation } from '../../__mocks__/@react-navigation/native'
import { StoreProvider, defaultState } from '../../src/contexts/store'
import ListContacts from '../../src/screens/ListContacts'
import { BasicAppContext, CustomBasicAppContext } from '../helpers/app'
import { TOKENS } from '../../src/container-api'
import { MainContainer } from '../../src/container-impl'

// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => {})
jest.useFakeTimers({ legacyFakeTimers: true })

const navigation = testUseNavigation()

describe('ListContacts Screen', () => {
  beforeEach(() => {
    jest.clearAllTimers()
    jest.clearAllMocks()
  })

  test('Renders correctly', async () => {
    const tree = render(
      <BasicAppContext>
        <ListContacts navigation={navigation as any} />
      </BasicAppContext>
    )

    await waitFor(() => {})

    expect(tree).toMatchSnapshot()
  })

  test('Pressing on a contact in the list takes the user to a contact history screen', async () => {
    const { findByText } = render(
      <BasicAppContext>
        <ListContacts navigation={navigation as any} />
      </BasicAppContext>
    )

    await waitFor(async () => {
      const connectionRecord = await findByText('BestBC Tea', { exact: true })
      fireEvent(connectionRecord, 'press')

      expect(navigation.navigate).toBeCalledWith('Contacts Stack', {
        screen: 'Chat',
        params: {
          connectionId: 'c426f2dc-0ffc-4252-b7d6-2304755f84d9',
        },
      })
    })
  })

  test('Hide list filters out specific contacts', async () => {
    const context = new MainContainer(container.createChildContainer()).init()
    const config = context.resolve(TOKENS.CONFIG)
    context.container.registerInstance(TOKENS.CONFIG, { ...config, contactHideList: ['Faber'] })
    const tree = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            developerModeEnabled: false,
          },
        }}
      >
        <CustomBasicAppContext container={context}>
          <ListContacts navigation={navigation as any} />
        </CustomBasicAppContext>
      </StoreProvider>
    )

    await waitFor(async () => {
      const teaContact = await tree.queryByText('BestBC Tea', { exact: true })
      expect(teaContact).toBe(null)
    })
  })

  test('Hide list does not filter out specific contacts when developer mode is enabled', async () => {
    const context = new MainContainer(container.createChildContainer()).init()
    const config = context.resolve(TOKENS.CONFIG)
    context.container.registerInstance(TOKENS.CONFIG, { ...config, contactHideList: ['Faber'] })
    const tree = render(
      <StoreProvider
        initialState={{
          ...defaultState,
          preferences: {
            ...defaultState.preferences,
            developerModeEnabled: true,
          },
        }}
      >
        <CustomBasicAppContext container={context}>
          <ListContacts navigation={navigation as any} />
        </CustomBasicAppContext>
      </StoreProvider>
    )

    await waitFor(async () => {
      const teaContact = await tree.queryByText('BestBC Tea', { exact: true })
      expect(teaContact).toBe(null)
    })
  })
})
