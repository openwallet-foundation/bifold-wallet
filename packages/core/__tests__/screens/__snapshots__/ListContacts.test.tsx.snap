// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContacts Screen Renders correctly 1`] = `
<View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "_tags": Object {},
          "alias": undefined,
          "autoAcceptConnection": undefined,
          "connectionTypes": Array [],
          "createdAt": "2024-09-04T18:50:28.647Z",
          "did": "did:peer:1zQmZYcmJeMX5Lc2HvQjf4VXSC3raHfBci6A4MnjDNCZTVp9",
          "errorMessage": undefined,
          "id": "b4b1b9cd-a445-4bb3-9645-a2377471965a",
          "imageUrl": undefined,
          "invitationDid": undefined,
          "mediatorId": undefined,
          "metadata": Object {},
          "outOfBandId": "b8aaa6fe-47c9-4ed8-8cb9-96299e4e0109",
          "previousDids": Array [],
          "previousTheirDids": Array [],
          "protocol": undefined,
          "role": "requester",
          "state": "completed",
          "theirDid": "did:peer:1zQmaiE6oyFzYYdBtKD3fJW6yfXE1r5hXMLfa7Ve8nundki6",
          "theirLabel": "BestBC College",
          "threadId": "5c667314-6465-4fa8-9a45-f66e571e99e2",
        },
        Object {
          "_tags": Object {},
          "alias": undefined,
          "autoAcceptConnection": undefined,
          "connectionTypes": Array [],
          "createdAt": "2024-09-04T18:50:28.647Z",
          "did": "did:peer:1zQmZYcmJeMX5Lc2HvQjf4VXSC3raHfBci6A4MnjDNCZTVp9",
          "errorMessage": undefined,
          "id": "3712d956-7731-428b-bcbb-127c0f6d615d",
          "imageUrl": undefined,
          "invitationDid": undefined,
          "mediatorId": undefined,
          "metadata": Object {},
          "outOfBandId": "d1036d48-4b88-4f63-9d7e-b4b0476f8108",
          "previousDids": Array [],
          "previousTheirDids": Array [],
          "protocol": undefined,
          "role": "requester",
          "state": "completed",
          "theirDid": "did:peer:1zQmaiE6oyFzYYdBtKD3fJW6yfXE1r5hXMLfa7Ve8nundki6",
          "theirLabel": "BestBC College",
          "threadId": "8df1d523-5415-4e62-9975-4d248fcb8f4a",
        },
        Object {
          "_tags": Object {},
          "alias": undefined,
          "autoAcceptConnection": undefined,
          "connectionTypes": Array [],
          "createdAt": "2024-09-04T18:50:28.647Z",
          "did": "did:peer:1zQmZYcmJeMX5Lc2HvQjf4VXSC3raHfBci6A4MnjDNCZTVp9",
          "errorMessage": undefined,
          "id": "c426f2dc-0ffc-4252-b7d6-2304755f84d9",
          "imageUrl": undefined,
          "invitationDid": undefined,
          "mediatorId": undefined,
          "metadata": Object {},
          "outOfBandId": "6e739679-db69-4228-9fdf-d1b8cc2431aa",
          "previousDids": Array [],
          "previousTheirDids": Array [],
          "protocol": undefined,
          "role": "requester",
          "state": "completed",
          "theirDid": "did:peer:1zQmaiE6oyFzYYdBtKD3fJW6yfXE1r5hXMLfa7Ve8nundki6",
          "theirLabel": "BestBC Tea",
          "threadId": "fc7405e5-039c-4b7a-bc3d-f9626fc96d25",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#313132",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="ContactDetails.AContact"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="com.ariesbifold:id/Contact"
        >
          <View
            style={
              Object {
                "backgroundColor": "#313132",
                "flexDirection": "row",
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FFFFFF",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                maxFontSizeMultiplier={2}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                B
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingVertical": 4,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                          "fontWeight": "bold",
                        },
                        undefined,
                      ]
                    }
                  >
                    BestBC College
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "paddingVertical": 4,
                    }
                  }
                />
              </View>
              <View />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#000000",
              "height": 1,
              "marginHorizontal": 16,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="ContactDetails.AContact"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="com.ariesbifold:id/Contact"
        >
          <View
            style={
              Object {
                "backgroundColor": "#313132",
                "flexDirection": "row",
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FFFFFF",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                maxFontSizeMultiplier={2}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                B
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingVertical": 4,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                          "fontWeight": "bold",
                        },
                        undefined,
                      ]
                    }
                  >
                    BestBC College
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "paddingVertical": 4,
                    }
                  }
                />
              </View>
              <View />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#000000",
              "height": 1,
              "marginHorizontal": 16,
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="ContactDetails.AContact"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="com.ariesbifold:id/Contact"
        >
          <View
            style={
              Object {
                "backgroundColor": "#313132",
                "flexDirection": "row",
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#FFFFFF",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 50,
                  "justifyContent": "center",
                  "marginRight": 16,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                maxFontSizeMultiplier={2}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 21,
                      "fontWeight": "bold",
                    },
                    Object {
                      "textAlign": "center",
                    },
                  ]
                }
              >
                B
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "paddingVertical": 4,
                    }
                  }
                >
                  <Text
                    maxFontSizeMultiplier={2}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                          "fontWeight": "bold",
                        },
                        undefined,
                      ]
                    }
                  >
                    BestBC Tea
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "paddingVertical": 4,
                    }
                  }
                />
              </View>
              <View />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
